src/app/Trade.ts

```diff
@@ -38,47 +38,81 @@ export class Trade {
   }
 
   private areOffersInvalid(
-    [animal, count]: Offer,
+    offer: Offer[],
     playerHerd: Herd,
-    [bankAnimal, bankCount]: Offer,
+    target: Offer[],
   ): boolean {
     return (
-      (count > 1 && bankCount > 1) ||
-      playerHerd.getAnimalNumber(animal) < count ||
-      this.bank.theHerd.getAnimalNumber(bankAnimal) < bankCount
+      (offer.length > 1 && target.length > 1) ||
+      this.isOfferQuantityInvalid(offer, playerHerd) ||
+      this.isOfferQuantityInvalid(target, this.bank.theHerd) ||
+      !this.isSingleQuantityOnAnySide(offer, target)
```

Comment:
perhaps we could have all the conditions either false or true (all is...Invalid or all !is....Valid)

Reply:
done